packages/core/src/types.ts

```diff
@@ -366,53 +366,51 @@ export type TransitionsConfig<TContext, TEvent extends EventObject> =
   | TransitionsConfigMap<TContext, TEvent>
   | TransitionsConfigArray<TContext, TEvent>;
 
-export type InvokeConfig<TContext, TEvent extends EventObject> =
-  | {
-      /**
-       * The unique identifier for the invoked machine. If not specified, this
-       * will be the machine's own `id`, or the URL (from `src`).
-       */
-      id?: string;
-      /**
-       * The source of the machine to be invoked, or the machine itself.
-       */
-      src:
-        | string
-        | StateMachine<any, any, any>
-        | InvokeCreator<TContext, TEvent, any>;
-      /**
-       * If `true`, events sent to the parent service will be forwarded to the invoked service.
-       *
-       * Default: `false`
-       */
-      autoForward?: boolean;
-      /**
-       * @deprecated
-       *
-       *  Use `autoForward` property instead of `forward`. Support for `forward` will get removed in the future.
-       */
-      forward?: boolean;
-      /**
-       * Data from the parent machine's context to set as the (partial or full) context
-       * for the invoked child machine.
-       *
-       * Data should be mapped to match the child machine's context shape.
-       */
-      data?: Mapper<TContext, TEvent> | PropertyMapper<TContext, TEvent>;
-      /**
-       * The transition to take upon the invoked child machine reaching its final top-level state.
-       */
-      onDone?:
-        | string
-        | SingleOrArray<TransitionConfig<TContext, DoneInvokeEvent<any>>>;
-      /**
-       * The transition to take upon the invoked child machine sending an error event.
-       */
-      onError?:
-        | string
-        | SingleOrArray<TransitionConfig<TContext, DoneInvokeEvent<any>>>;
-    }
-  | StateMachine<any, any, any>;
```

Comment:
I've moved this out of `InvokeConfig` to keep `InvokeConfig` focused on the object description of the invocable actor - it's still allowed, just defined directly on the `invoke` property of `StateNodeConfig`:
https://github.com/davidkpiano/xstate/pull/1262/files#diff-2991521f751aa9cf68976ceef512e19aR463